flink-core/src/main/java/org/apache/flink/configuration/SecurityOptions.java

```diff
@@ -299,6 +300,26 @@
 			.defaultValue(true)
 			.withDescription("Flag to enable peer’s hostname verification during ssl handshake.");
 
+	/**
+	 * SSL engine provider.
+	 */
+	public static final ConfigOption<String> SSL_PROVIDER =
+		key("security.ssl.provider")
+			.defaultValue("JDK")
+			.withDescription(Description.builder()
+					.text("The SSL engine provider to use for the ssl transport:")
+					.list(
+						TextElement.text("%s: default Java-based SSL engine", TextElement.code("JDK")),
+						TextElement.text("%s: openSSL-based SSL engine using system libraries"
+							+ " (falls back to JDK if not available)", TextElement.code("OPENSSL"))
```

Comment:
It would be safer to fail instead of fall back if users specifies the openSSL explicitly.

Reply:
Do you think, it would then be a good idea to also have `AUTO` with the behaviour that it chooses openSSL if available, otherwise Java SSL?

Reply:
It may be dangerous though if this would result in a heterogeneous cluster where some nodes had to fall back to JavaSSL while others are using openSSL. Maybe let's leave it with the user having to decide manually.

Reply:
That is a good question. I think an `AUTO` mode (even as the default value) would make sense. But I leave it up to you to decide whether this is worth the extra effort or not.

Problem in non heterogeneous shouldn't be that frequent. I was mainly concerned with the situation where somebody is explicitly setting the value to `OpenSSL` and it is silently not being respected.

Reply:
I'll leave it without `AUTO` since, especially with standalone clusters, it may actually quite simple to accidentally get into a heterogeneous SSL setup by missing some dependencies on some nodes. I fear that debugging such scenarios would be painful